### PR TITLE
Feature/2q24 57

### PR DESCRIPTION
**Pull request by chriscampos20**

### CODE REVIEW

File: src/periods/dto/create-period.dto.ts [hunk 1]
```diff
@@ -2,8 +2,13 @@ import { ApiProperty } from '@nestjs/swagger';
...
   name: string;
+
   @ApiProperty()
   start: Date;
+
   @ApiProperty()
   end: Date;
+
+  @ApiProperty()
+  search?: string;
```
Comment (wmartzh) [inline src/periods/dto/create-period.dto.ts:13]: This is not needed here

File: src/periods/periods.controller.ts [hunk 2]
```diff
@@ -6,18 +6,27 @@ import { GetPeriodDto } from './dto/get-period.dto';
...
     return this.periodsService.findAll(query);
   }
+  @Get('search')
+  async search(@Query('search') search?: string): Promise<Period[]> {
+    const query: CreatePeriodDto = {
+      search,
+      name: '',
+      start: undefined,
+      end: undefined
+    };
```
Comment (wmartzh) [inline src/periods/periods.controller.ts:27]: Create dto is nor for search

File: src/periods/periods.service.ts [hunk 3]
```diff
@@ -28,6 +28,18 @@ export class PeriodsService {
     }
   }
 
+  async search(query: CreatePeriodDto): Promise<Period[]> {
```
Comment (wmartzh) [inline src/periods/periods.service.ts:31]: the search param must be included into the get all search

Comment (chriscampos20) [reply src/periods/periods.service.ts]: async findAll(query: CreatePeriodDto): Promise<PaginatedResponse<Period>> {
    const { page = 1, size = 10, search } = query;
    const { take, skip } = createPaginationMetadata(page, size);
    const prismaQuery = {
      take,
      skip,
      where: search ? { name: { contains: search } } : {},
    };
    const [period, total] = await Promise.all([
      this.prismaService.period.findMany(prismaQuery),
      this.prismaService.period.count(),
    ]);
    return createPaginatedResponse<Period>(period, total, page, size);
  }

File: src/periods/periods.service.ts [hunk 4]
```diff
@@ -28,13 +27,13 @@ export class PeriodsService {
     }
   }
 
-  async findAll(query: GetPeriodDto): Promise<PaginatedResponse<Period>> {
-    const { page = 1, size = 10 } = query;
+  async findAll(query: CreatePeriodDto): Promise<PaginatedResponse<Period>> {
```
Comment (wmartzh) [inline src/periods/periods.service.ts:30]: This should not use CreateDTO

File: src/periods/dto/update-period.dto.ts [hunk 5]
```diff
@@ -1,4 +1,8 @@
 import { PartialType } from '@nestjs/swagger';
 import { CreatePeriodDto } from './create-period.dto';
 
-export class UpdatePeriodDto extends PartialType(CreatePeriodDto) {}
+export class UpdatePeriodDto extends PartialType(CreatePeriodDto) {
+  name: any;
+  start: any;
+  end: any;
```
Comment (wmartzh) [inline src/periods/dto/update-period.dto.ts:7]: Delete this

File: src/periods/dto/create-period.dto.ts [hunk 6]
```diff
@@ -1,9 +1,16 @@
...
-  @ApiProperty()
-  name: string;
-  @ApiProperty()
-  start: Date;
-  @ApiProperty()
-  end: Date;
+import { PaginationQueryDto } from '../../utils/pagination.util';
+import { ApiPropertyOptional } from '@nestjs/swagger';
+
+export class CreatePeriodDto extends PaginationQueryDto {
```
Comment (wmartzh) [inline src/periods/dto/create-period.dto.ts:4]: CreatePeriodDto must use for create endpoint, not for get

**Pull request closed without merging**